src/ha-utils/ha/gcp/providers/GCPIAAS.go

```diff
@@ -0,0 +1,361 @@
+
+	for _, currentNetworkInterface := range virtualMachine.NetworkInterfaces {
+		if currentNetworkInterface.NetworkIP == iaasProvider.Config.CurrentInstanceIP {
+			network = currentNetworkInterface.Network
+			subNetwork = currentNetworkInterface.Subnetwork
+			break
+		}
+	}
+	return region, regionName, availabilityZone, azName, network, subNetwork
+
+}
+
+func (iaasProvider *GCPIAAS) getLoadBalancerName() string {
+	return "lb-" + iaasProvider.Config.DeploymentGuid
+}
+
+func (iaasProvider *GCPIAAS) getVMGroupName() string {
+	return iaasProvider.getLoadBalancerName() + "-" + strings.Replace(iaasProvider.Config.CurrentInstanceIP, ".", "-", -1)
+}
+
+func (iaasProvider *GCPIAAS) getHealhProbeName() string {
+	return "health-check--" + iaasProvider.getLoadBalancerName()
+}
+
+func (iaasProvider *GCPIAAS) getLoadBalancingRuleName() string {
+	return "lbrule-" + iaasProvider.getLoadBalancerName()
+}
+
+func (iaasProvider *GCPIAAS) ManageResources() int {
+
+	var loadBalancerName, vmGroupName, healthProbeName, lbRuleName, regionName, availabilityZone, azName, network, subNetwork string
+	var virtualMachine *gcpmodels.VirtualMachine
+	var vmGroup *gcpmodels.VMGroup
+	var probe *gcpmodels.Probe
+	var loadBalancer *gcpmodels.LoadBalancer
+	var lbRule *gcpmodels.LoadBalancingRule
+	var returnValue bool
+
+	//	MANAGE INTERNAL LOAD BALANCER
+	loadBalancerName = iaasProvider.getLoadBalancerName()
+	log.Println("load balancer name is ", loadBalancerName)
+
+	// Lets identify the virtual machine and its related details.
+	virtualMachine, returnValue = iaasProvider.vmService.GetVirtualMachineByIP(iaasProvider.Config.CurrentInstanceIP)
+	if returnValue == false || virtualMachine == nil {
+		log.Println("Failed to retrieve virtual machine details - ", virtualMachine.Name)
+		return 1
+	}
+
+	_, regionName, availabilityZone, azName, network, subNetwork = iaasProvider.initializeLandscapeInfo(virtualMachine)
+
+	// Identify whether a instance group exists for this instance.
+	vmGroupName = iaasProvider.getVMGroupName()
+	vmGroup, returnValue = iaasProvider.vmGroupService.GetVMGroup(vmGroupName, azName)
+	if returnValue == false {
+		log.Println("Failed to retrieve vm group details - ", vmGroupName)
+		return 2
+	} else if returnValue == true && vmGroup != nil {
+		log.Println("VM group with name:", vmGroupName, "already exists in GCP.")
+	}
+	// If the vm group doesn't exist, then create one.
+	if vmGroup == nil {
+		log.Println("VM group with name ", vmGroupName, "does not exist in GCP. Initiating creation of VM group.")
+		returnValue = iaasProvider.vmGroupService.CreateVMGroup(vmGroupName, availabilityZone, network, subNetwork)
+		if returnValue == false {
+			return 3
+		}
+		vmGroup, returnValue = iaasProvider.vmGroupService.GetVMGroup(vmGroupName, azName)
+		if returnValue == false {
+			return 5
+		}
+	}
+	// Let's add this instance to vm group
+	returnValue = iaasProvider.vmGroupService.AddVMToVMGroup(vmGroupName, azName, virtualMachine.SelfLink)
```

Comment:
is this method `manageResources` invoked twice? I dont see two vms added to the group? Only one VM is being added?

Reply:
This code runs on both the VMs of the deployment. It adds the vm it is running on.